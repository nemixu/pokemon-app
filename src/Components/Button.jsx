import React, { useState } from 'react';

function Button({ label, isDisabled, className, handleClick }) {

	const [clicked, setClicked] = useState(0);

	const localHandleClick = () => {
		setClicked(clicked+1)

		//pass handleclick as jest mock function, asserrt that it was called number of times X button was clicked.
		handleClick();
	}

	return (
		<>
		<p>Clicked {clicked}</p>
		<button
			type="button"
			className={className}
			onClick={localHandleClick}
			disabled={isDisabled }>
			{isDisabled ? "disabled" : label }
		</button>
		</>
		
	);
}

export default Button;
