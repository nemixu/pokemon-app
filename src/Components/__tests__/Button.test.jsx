import Button from "../Button";
import {render, screen, fireEvent} from "@testing-library/react";
import '@testing-library/jest-dom'

//rendering the button with screen.
describe("Button.jsx", () => {
    it("should render correctly", () => {
        render(<Button label="test" className="colour"/>);
        const button = screen.getByRole("button");
        // expect(screen.getByText("test")).toBeInTheDocument();
        // expect(button).toHaveClass("colour");
        expect(button).toMatchSnapshot();

    });

    it.skip("Should be rendered as disabled", ()=> {
        render(<Button isDisabled />);
        const button = screen.getByRole("button");
        expect(button).toBeDisabled();
        screen.getByText("disabled");
        //expect(button).toHaveTextContent("disabled");
    })

    it("should increment the click value by 1", () => {
        //arrange
        const mockClickHandler = jest.fn();
        render(<Button handleClick={mockClickHandler}/> );
        const button = screen.getByRole("button");
        //act
        fireEvent.click(button);
        fireEvent.click(button);
        fireEvent.click(button);
        //assert
        expect(screen.getByText("Clicked 3")).toBeInTheDocument();
        expect(mockClickHandler).toHaveBeenCalledTimes(3);
    

    })


})