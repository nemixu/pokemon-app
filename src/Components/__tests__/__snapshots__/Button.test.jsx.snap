// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.jsx should render correctly 1`] = `
<button
  class="colour"
  type="button"
>
  test
</button>
`;
